### PR TITLE
[spirv-tools] Add usage, use lower-case config dirs

### PR DESCRIPTION
**Pull request by dg0yt**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/spirv-tools/usage [hunk 1]
```diff
@@ -0,0 +1,19 @@
+spirv-tools provides CMake targets:
+
+    find_package(SPIRV-Tools CONFIG REQUIRED)
+    # Note: SPIRV-Tools provides two targets.
+    # Only SPIRV-Tools-static has full symbol visibility.
+    target_link_libraries(main PRIVATE $<IF:$<BOOL:${BUILD_SHARED_LIBS}>,SPIRV-Tools-shared,SPIRV-Tools-static>)
```
Comment (dg0yt) [inline ports/spirv-tools/usage:6]: This PR is meant to document the role of these targets. ATM, usage heuristics proposed to link both (wrong).
Upstream doesn't want to sort this out yet (https://github.com/KhronosGroup/SPIRV-Tools/issues/3909), and neither do I.

Comment (LilyWangLL) [reply ports/spirv-tools/usage:6]: The `usage` test failed with the following error:
```
1> [CMake] CMake Error at E:/PRreview/spirv/scripts/buildsystems/vcpkg.cmake:853 (_find_package):
1> [CMake]   Found package configuration file:
1> [CMake] 
1> [CMake]     E:/PRreview/spirv/installed/x64-windows/share/spirv-tools-link/SPIRV-Tools-linkConfig.cmake
1> [CMake] 
1> [CMake]   but it set SPIRV-Tools-link_FOUND to FALSE so package "SPIRV-Tools-link" is
1> [CMake]   considered to be NOT FOUND.  Reason given by package:
1> [CMake] 
1> [CMake]   The following imported targets are referenced, but are missing:
1> [CMake]   SPIRV-Tools-opt
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeProject16/CMakeLists.txt:17 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
```
Are this also upstream issues?

Comment (dg0yt) [reply ports/spirv-tools/usage:6]: I will check. But I guess this is vcpkg trouble. The usage file is derived from a heuristics. It is not a good starting point.

Comment (dg0yt) [reply ports/spirv-tools/usage:6]: It is an upstream issue. But the vcpkg is lagging behind, and I don't want to deal with an update now.

Note: It is not a regression from this PR.

**Comment by dg0yt** [comment]
Integrated into #31912.

**Pull request closed without merging**